scipy/optimize/_sda.py

```diff
@@ -0,0 +1,572 @@
+    maxiter : int, optional
+        The maximum number of sda iterations. Increase this value if the
+        objective function is very complicated with high dimensions.
+    local_search_options : dict, optional
+        Extra keyword arguments to be passed to the local minimizer
+            ``scipy.optimize.minimize()`` Some important options could be:
+            method : str
+                The minimization method (e.g. ``"L-BFGS-B"``)
+            args : tuple
+                Extra arguments passed to the objective function (``func``) and
+                its derivatives (Jacobian, Hessian).
+    initial_temp : float, optional
+        The initial temperature, use higher values to facilitates a wider
+        search of the energy landscape, allowing sda to escape local minima
+        that it is trapped in.
+    visit : float, optional
+        Parameter for visiting distribution. Higher values give the visiting
+        distribution a heavier tail, this makes the algorithm jump to a more
+        distant region. The value range is (0, 3]
+    accept : float, optional
+        Parameter for acceptance distribution. It is used to control the
+        probability of acceptance. The lower the acceptance parameter, the
+        smaller the probability of acceptance. It has to be any negative value.
+    maxfun : int, optional
+        Soft limit for the number of objective function calls. If the
+        algorithm is in the middle of a local search, this number will be
+        exceeded, the algorithm will stop just after the local search is
+        done.
+    seed : int or `np.random.RandomState`, optional
+        If `seed` is not specified the `np.RandomState` singleton is used.
+        If `seed` is an int, a new `np.random.RandomState` instance is used,
+        seeded with seed.
+        If `seed` is already a `np.random.RandomState instance`, then that
+        `np.random.RandomState` instance is used.
+        Specify `seed` for repeatable minimizations. The random numbers
+        generated with this seed only affect the visiting distribution
+        function and new coordinates generation.
+    no_local_search : boolean, optional
+        If `no_local_search` is set to `True`, a traditional Generalized
+        Simulated Annealing will be performed with no local search
+        strategy applied.
+
+    Returns
+    -------
+    res : OptimizeResult
+        The optimization result represented as a ``OptimizeResult`` object.
+        Important attributes are: ``x`` the solution array, ``fun`` the value
+        of the function at the solution, and ``message`` which describes the
+        cause of the termination.
+        See `OptimizeResult` for a description of other attributes.
+
+    Notes
+    -----
+    SDA is an implementation of the Simulated Dual Annealing. This stochastic
+    approach [3]_ implements the generalization of CSA (Classical Simulated
```

Comment:
Is there a set of parameters to do only classical simulated annealing?  I think many people would find that useful, at the very least for comparison.  It might be nice to add an example below for that or specify how in the documentation.

Reply:
There is no way for user to do either CSA of FSA but by using no_local_search option to True, GSA (Generalized Simulated Annealing) will be applied. Maybe this can be said in the argument docstring. I'll do that.